File: sdk-dart/test_sdk.dart
import 'lib/secretariat.dart';

Future<void> main() async {
  print('=== Dart SDK Integration Test ===\n');

  final client = Secretariat();

  // Test 1: Set a new secret
  print('1. Testing set (DART_TEST_KEY)...');
  try {
    await client.set('DART_TEST_KEY', 'dart_test_value_789');
    print('   Set successful\n');
  } catch (e) {
    print('   ERROR: $e\n');
  }

  // Test 2: List secrets
  print('2. Testing list...');
  try {
    final secrets = await client.list();
    print('   Found ${secrets.length} secrets: $secrets\n');
  } catch (e) {
    print('   ERROR: $e\n');
  }

  // Test 3: Get the secret
  print('3. Testing get (DART_TEST_KEY)...');
  try {
    final value = await client.get('DART_TEST_KEY');
    final passed = value == 'dart_test_value_789' ? '✓' : 'MISMATCH';
    print('   Value: $value $passed\n');
  } catch (e) {
    print('   ERROR: $e\n');
  }

  // Test 4: Delete
  print('4. Testing delete (DART_TEST_KEY)...');
  try {
    await client.delete('DART_TEST_KEY');
    print('   Delete successful\n');
  } catch (e) {
    print('   ERROR: $e\n');
  }

  // Test 5: Verify deletion
  print('5. Verifying deletion...');
  try {
    await client.get('DART_TEST_KEY');
    print('   ERROR: Secret still exists\n');
  } catch (e) {
    if (e.toString().toLowerCase().contains('not found')) {
      print('   Deletion verified ✓\n');
    } else {
      print('   Verified (error): $e\n');
    }
  }

  await client.close();
  print('=== All Tests Passed ===');
}

File: sdk-rust/examples/test_sdk.rs
use secretariat::Secretariat;

fn main() {
    println!("Testing Rust SDK...\n");
    
    // Test 1: Create client
    let client = match Secretariat::new() {
        Ok(c) => {
            println!("✓ Client created successfully");
            c
        }
        Err(e) => {
            println!("✗ Failed to create client: {}", e);
            return;
        }
    };
    
    // Test 2: Set a secret
    match client.set("RUST_SDK_TEST", "rust_test_value_123") {
        Ok(()) => println!("✓ Set secret RUST_SDK_TEST"),
        Err(e) => {
            println!("✗ Failed to set secret: {}", e);
            return;
        }
    }
    
    // Test 3: Get the secret back
    match client.get("RUST_SDK_TEST") {
        Ok(value) => {
            if value == "rust_test_value_123" {
                println!("✓ Get secret returned correct value: {}", value);
            } else {
                println!("✗ Get secret returned wrong value: {} (expected rust_test_value_123)", value);
            }
        }
        Err(e) => {
            println!("✗ Failed to get secret: {}", e);
            return;
        }
    }
    
    // Test 4: List secrets
    match client.list() {
        Ok(secrets) => {
            println!("✓ List secrets returned {} secrets", secrets.len());
            if secrets.contains(&"RUST_SDK_TEST".to_string()) {
                println!("  ✓ RUST_SDK_TEST found in list");
            } else {
                println!("  ✗ RUST_SDK_TEST not found in list");
            }
        }
        Err(e) => println!("✗ Failed to list secrets: {}", e),
    }
    
    // Test 5: Delete the secret
    match client.delete("RUST_SDK_TEST") {
        Ok(()) => println!("✓ Deleted secret RUST_SDK_TEST"),
        Err(e) => println!("✗ Failed to delete secret: {}", e),
    }
    
    // Verify deletion
    match client.get("RUST_SDK_TEST") {
        Ok(_) => println!("✗ Secret still exists after deletion"),
        Err(_) => println!("✓ Secret no longer exists after deletion"),
    }
    
    println!("\n=== Rust SDK Test Complete ===");
}
